service/src/main/java/bio/terra/cbas/dependencies/leonardo/AppUtils.java

```diff
@@ -34,21 +35,25 @@ public AppUtils(
 
   int appComparisonFunction(ListAppResponse a, ListAppResponse b) {
 
-    // First criteria: Prefer apps with the WDS app type.
+    // First criteria: Prefer apps with the expected app type.
     // NB: Negative because lower index is better
-    int appTypeScoreA = -leonardoServerConfiguration.wdsAppTypes().indexOf(a.getAppType());
-    int appTypeScoreB = -leonardoServerConfiguration.wdsAppTypes().indexOf(b.getAppType());
+    int appTypeScoreA = -leonardoServerConfiguration.appTypeNames().indexOf(a.getAppType());
+    int appTypeScoreB = -leonardoServerConfiguration.appTypeNames().indexOf(b.getAppType());
     if (appTypeScoreA != appTypeScoreB) {
       return appTypeScoreA - appTypeScoreB;
     }
 
-    // Second criteria: Prefer apps with the WDS name
+    // Second criteria: Prefer apps with the expected app type name name
     int nameScoreA =
-        Objects.equals(a.getAppName(), "wds-%s".formatted(wdsServerConfiguration.instanceId()))
+        Objects.equals(
+                a.getAppName(),
+                "%s-%s".formatted(a.getAppType(), wdsServerConfiguration.instanceId()))
```

Comment:
I think the `"%s-%s".formatted(a.getAppType(), wdsServerConfiguration.instanceId())` check might not work for Cromwell apps because for Cromwell/CBAS apps, the name of app follows the pattern `terra-app-<random-uuid>`. It might be worth testing out these changes against an a running Cromwell app in Terra dev.